### ISSUE TITLE
ROCm

### ISSUE THREAD

**Issue by etemiz** [VikParuchuri/marker#217]
Thanks for this amazing software!

Is there a possibility to run it or surya using ROCm / AMD GPUs? 

Thanks again

**Comment by etemiz** [comment]
I found it: 

```
pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm6.0
pip install marker-pdf
```

**Comment by mxchinegod** [comment]
@etemiz are you seeing models load into VRAM i.e. using 7900XTX and ROCm 6.0, I'm only seeing 

```
Loaded detection model vikp/surya_det2 on device cpu with dtype torch.float32
Loaded detection model vikp/surya_layout2 on device cpu with dtype torch.float32
Loaded reading order model vikp/surya_order on device cpu with dtype torch.float32
Loaded recognition model vikp/surya_rec on device cpu with dtype torch.float32
Loaded texify model to cpu with torch.float32 dtype
```

I was looking into adding ROCm support in PR but if you have it working already I wonder what I am doing incorrectly.

**Comment by etemiz** [comment]
Yes I see them in the VRAM.

**Comment by sgarchavada** [comment]
Yes — Surya runs on AMD GPUs via ROCm. I got it working on an MI300X with no code changes.

**Setup**
- ROCm PyTorch (I used torch 2.9 from a ROCm image). Surya's device auto-detect just works, because ROCm exposes the GPU through the `torch.cuda` / HIP API — `torch.cuda.is_available()` returns `True`, so it selects the AMD GPU. You can also force `TORCH_DEVICE=cuda`.
- No CUDA-only deps required (xformers / bitsandbytes / triton not needed) — just torch + transformers + the usual pure-python packages.
- detection / foundation / recognition / layout all load (~2 s) with zero HIP errors, and recognized text matches CUDA on clean pages. (On low-quality scans the two architectures' kernels can disagree on a handful of hard glyphs — expected floating-point divergence across GPUs; it doesn't affect clean text, and the extracted numbers were ~95% preserved in my tests.)

**Correcting my earlier note on flash-attn + speed**, after more testing:

- flash-attn-2 actually *does* build and run on ROCm (CDNA — MI200 / MI300). The only thing gating it is Surya's `is_flash_attn_2_supported()`, which checks `torch.version.cuda` (which is `None` on ROCm). Adding one branch — accept `torch.version.hip` — enables it, and flash-attn then runs cleanly on the MI300X.
- **But it does not close the speed gap — it's speed-neutral for OCR.** I expected it to help; it doesn't. Recognition is eager-mode and autoregressive, so it's **latency-bound, not attention-compute-bound**: during OCR the GPU sits at ~0–7% utilization, and the bottleneck is per-op kernel-launch latency, not the attention math. Speeding up attention therefore changes nothing.
- Speed reality: warm single small images are ~1.5 s, but on real multi-line document pages the MI300X is meaningfully slower than an NVIDIA card running the same models. The real lever is cutting launch overhead — `torch.compile` / CUDA-graph capture / batched multi-request serving — not flash-attn.

Net: **correctness on ROCm is solid**; OCR throughput is currently limited by eager-mode latency rather than the attention kernel.
